wcmc_components/app/controllers/wcmc_components/table_controller.rb

```diff
@@ -1,31 +1,39 @@
 module WcmcComponents
   class TableController < ApplicationController
+    def index
+      @results = table_class.paginate_for_table(**table_params_with_symbol_keys)
 
+      render json: @results
+    end
+
+    private
+
+    # Returns the class that corresponds to this table
     def table_class
-      # walk up url to find which class table we are in
+      # Get the path
       path = request.fullpath
-      while !path.nil?
-        if !WcmcComponents.routes_classes[path].nil?
-          return WcmcComponents.routes_classes[path].constantize
-        elsif !path.rindex('/').nil?
-          path = path[0, path.rindex('/')+1]
-        else
-          return nil
-        end
-      end
-    end
-    
-    def list
-      @results = table_class.paginate(params.to_json)
-      render json: @results
+
+      # Get all registered paths which "match" our path. E.g., the ones that `path` begins with
+      matching_paths = WcmcComponents.routes_classes.keys.select { |defined_route| path.starts_with? defined_route }
+
+      # The longest of those is the most precise match.
+      # Get the value it stores in routes_classes and return it as a constant
+      WcmcComponents&.routes_classes&.dig(matching_paths.max)&.constantize
```

Comment:
Doesn't make much sense to use safe navigation here for WcmcComponents.route_classes and not on 17